src/main/java/com/doctusoft/gwtmock/Document.java

```diff
@@ -186,12 +197,23 @@ public void printFormatted(Element element, String indent, PrintWriter pw) {
 			pw.print(" " + attribute.getKey() + "=\"" + attribute.getValue() + "\"");
 		}
 		pw.println(">");
-		if (!Strings.isNullOrEmpty(element.getInnerText())) {
```

Comment:
Yes. The text nodes are now nodes in Simulated DOM so no special handling is required.

e.g.  setInnerText(text) does this:  .appendChild(document.createTextNode(text));